### ISSUE TITLE
Dimension coordinate with auto-numbering gives error

### ISSUE THREAD

**Issue by ivar-rummelhoff** [pymc-devs/nutpie#62]
If I provide `length` instead of `values` to `pymc.Model.add_chord`, I get the error below.
I assume this is a bug.

Kind regards,
Ivar Rummelhoff

```
TypeError                                 Traceback (most recent call last)
Cell In[6], line 2
      1 with model:
----> 2     idata = pm.sample(random_seed=9641478, nuts_sampler='nutpie')

File .venv/lib/python3.11/site-packages/pymc/sampling/mcmc.py:658, in sample(draws, tune, chains, cores, random_seed, progressbar, step, nuts_sampler, initvals, init, jitter_max_retries, n_init, trace, discard_tuned_samples, compute_convergence_checks, keep_warning_stat, return_inferencedata, idata_kwargs, nuts_sampler_kwargs, callback, mp_ctx, model, **kwargs)
    654     if not isinstance(step, NUTS):
    655         raise ValueError(
    656             "Model can not be sampled with NUTS alone. Your model is probably not continuous."
    657         )
--> 658     return _sample_external_nuts(
    659         sampler=nuts_sampler,
    660         draws=draws,
    661         tune=tune,
    662         chains=chains,
    663         target_accept=kwargs.pop("nuts", {}).get("target_accept", 0.8),
    664         random_seed=random_seed,
    665         initvals=initvals,
    666         model=model,
    667         progressbar=progressbar,
    668         idata_kwargs=idata_kwargs,
    669         nuts_sampler_kwargs=nuts_sampler_kwargs,
    670         **kwargs,
    671     )
    673 if isinstance(step, list):
    674     step = CompoundStep(step)

File .venv/lib/python3.11/site-packages/pymc/sampling/mcmc.py:297, in _sample_external_nuts(sampler, draws, tune, chains, target_accept, random_seed, initvals, model, progressbar, idata_kwargs, nuts_sampler_kwargs, **kwargs)
    291 if idata_kwargs is not None:
    292     warnings.warn(
    293         "`idata_kwargs` are currently ignored by the nutpie sampler",
    294         UserWarning,
    295     )
--> 297 compiled_model = nutpie.compile_pymc_model(model)
    298 idata = nutpie.sample(
    299     compiled_model,
    300     draws=draws,
   (...)
    306     **nuts_sampler_kwargs,
    307 )
    308 return idata

File .venv/lib/python3.11/site-packages/nutpie/compile_pymc.py:190, in compile_pymc_model(model, **kwargs)
    185 expand_numba_raw, c_sig_expand = _make_c_expand_func(
    186     n_dim, n_expanded, expand_fn, user_data, shared_expand, shared_data
    187 )
    188 expand_numba = numba.cfunc(c_sig_expand, **kwargs)(expand_numba_raw)
--> 190 coords = {name: pd.Index(vals) for name, vals in model.coords.items()}
    191 if "unconstrained_parameter" in coords:
    192     raise ValueError("Model contains invalid name 'unconstrained_parameter'.")

File .venv/lib/python3.11/site-packages/nutpie/compile_pymc.py:190, in <dictcomp>(.0)
    185 expand_numba_raw, c_sig_expand = _make_c_expand_func(
    186     n_dim, n_expanded, expand_fn, user_data, shared_expand, shared_data
    187 )
    188 expand_numba = numba.cfunc(c_sig_expand, **kwargs)(expand_numba_raw)
--> 190 coords = {name: pd.Index(vals) for name, vals in model.coords.items()}
    191 if "unconstrained_parameter" in coords:
    192     raise ValueError("Model contains invalid name 'unconstrained_parameter'.")

File .venv/lib/python3.11/site-packages/pandas/core/indexes/base.py:524, in Index.__new__(cls, data, dtype, copy, name, tupleize_cols)
    521         data = com.asarray_tuplesafe(data, dtype=_dtype_obj)
    523 elif is_scalar(data):
--> 524     raise cls._raise_scalar_data_error(data)
    525 elif hasattr(data, "__array__"):
    526     return Index(np.asarray(data), dtype=dtype, copy=copy, name=name)

File .venv/lib/python3.11/site-packages/pandas/core/indexes/base.py:5265, in Index._raise_scalar_data_error(cls, data)
   5260 @final
   5261 @classmethod
   5262 def _raise_scalar_data_error(cls, data):
   5263     # We return the TypeError so that we can raise it from the constructor
   5264     #  in order to keep mypy happy
-> 5265     raise TypeError(
   5266         f"{cls.__name__}(...) must be called with a collection of some "
   5267         f"kind, {repr(data) if not isinstance(data, np.generic) else str(data)} "
   5268         "was passed"
   5269     )

TypeError: Index(...) must be called with a collection of some kind, None was passed
```

**Comment by aseyboldt** [comment]
How do you provide a length? I didn't know you could do that, so maybe more of a missing feature than a bug. :-)

**Comment by ivar-rummelhoff** [comment]
I would say that the documentation for `add_cord` is pretty clear:
https://www.pymc.io/projects/docs/en/v5.8.0/api/model/generated/classmethods/pymc.model.core.Model.add_coord.html#pymc.model.core.Model.add_coord

As a workaround I have used `values=np.arange(n)` instead of `length=n`. This seems to result in a larger inference data object.

**Comment by aseyboldt** [comment]
Ah, sorry, I was confusing things with the `with_coords` in nutpie. 
Should be fixed by #65
